cryptography/src/bls12381/primitives/group.rs

```diff
@@ -79,28 +81,55 @@ const BLST_FR_ONE: Scalar = Scalar(blst_fr {
 #[repr(transparent)]
 pub struct G1(blst_p1);
 
+/// The size in bytes of an encoded G1 element.
 pub const G1_ELEMENT_BYTE_LENGTH: usize = 48;
 
+/// Domain separation tag for hashing a proof of possession (compressed G2) to G1.
+pub const DST_G1_PROOF_OF_POSSESSION: &[u8] = b"BLS_POP_BLS12381G1_XMD:SHA-256_SSWU_RO_POP_";
+
 /// Domain separation tag for hashing a message to G1.
-pub const DST_G1: &[u8] = b"BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_";
+pub const DST_G1_MESSAGE: &[u8] = b"BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_";
 
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 #[repr(transparent)]
 pub struct G2(blst_p2);
 
+/// The size in bytes of an encoded G2 element.
 pub const G2_ELEMENT_BYTE_LENGTH: usize = 96;
 
+/// Domain separation tag for hashing a proof of possession (compressed G1) to G2.
+pub const DST_G2_PROOF_OF_POSSESSION: &[u8] = b"BLS_POP_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_";
+
 /// Domain separation tag for hashing a message to G2.
-pub const DST_G2: &[u8] = b"BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_";
+pub const DST_G2_MESSAGE: &[u8] = b"BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_";
 
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 pub struct GT(blst_fp12);
 
+/// The private key type.
 pub type Private = Scalar;
+
+/// The private key length.
 pub const PRIVATE_KEY_LENGTH: usize = SCALAR_LENGTH;
+
+/// The public key type (G1).
 pub type Public = G1;
+
+/// The public key length (G1).
+pub const PUBLIC_KEY_LENGTH: usize = G1_ELEMENT_BYTE_LENGTH;
```

Comment:
I ended up defining a lot more constants for our "default" handling ("minimum PK") rather than exporting `G1`/`G2` to make basic usage easier.